ui/selectors/selectors.js

```diff
@@ -375,6 +376,17 @@ export function getAddressBook(state) {
   return Object.values(state.metamask.addressBook[chainId]);
 }
 
+export function getEnsResolutionByAddress(state, address) {
+  return state.metamask.ensResolutionsByAddress[address] || '';
+}
+
+export function getRecipientEnsName(state) {
+  const recipientAddress = getSendTo(state);
+  const checksummedAddress = toChecksumHexAddress(recipientAddress);
+  const recipientEns = getEnsResolutionByAddress(state, checksummedAddress);
+  return recipientEns;
+}
```

Comment:
Let's bake this functionality into the getRecipient selector in the send duck.

You can keep the genEnsResolutionByAddress selector you have here, but in the send duck modify the getRecipient selector such that it returns an ens resolution as the nickname if a nickname isn't present but and ens resolution is. Anywhere we display recipient.nickname then will benefit from this treatment.

Reply:
@brad-decker Fixed [here](https://github.com/MetaMask/metamask-extension/pull/14613/commits/b9250908d9228c50586ebd8b8a8c561864a12169). I used `state.metamask.ensResolutionsByAddress` in the `getRecipient` selector in the send duck to check if the ens resolusion is present. If you think this approach isn't right, I would like some new suggestion if you have about this.
Thank you in advance!